linkerd/transport-header/src/lib.rs

```diff
@@ -14,13 +14,21 @@ mod proto {
     include!(concat!(env!("OUT_DIR"), "/transport.l5d.io.rs"));
 }
 
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, PartialEq, Hash)]
 pub struct TransportHeader {
     /// The target port.
     pub port: u16,
 
     /// The logical name of the target (service), if one is known.
     pub name: Option<Name>,
+
+    pub protocol: Option<SessionProtocol>,
+}
+
+#[derive(Clone, Debug, PartialEq, Hash)]
+pub enum SessionProtocol {
+    Http1,
+    H2,
```

Comment:
nit/TIOLI: is there any particular reason we add a new `SessionProtocol` enum, versus just converting the protobuf session protocol enum to/from `http::Version` directly? that seems like it could be simpler.

i suppose we might want to add non-http session protocols in the future...but it might be simpler to just separate them _then_? i could go either way on this, it's just a thought.

Reply:
I don't think the transport-header module should have a dependency on proxy-http. i'd prefer to keep these things domain-specificon the assumption this will grow to include non-HTTP things.

Reply:
ah, yeah, that makes sense. carry on 👍